src/coreclr/vm/peassembly.cpp

```diff
@@ -1467,13 +1259,18 @@ TADDR PEFile::GetMDInternalRWAddress()
         // 3) ASSUMPTION: We are assuming that no pointer adjustment is required to convert between
         //    IMDInternalImport*, IMDInternalImportENC* and MDInternalRW*. Ideally I was hoping to do this with a
         //    static_cast<> but the compiler complains that the ENC<->RW is an unrelated conversion.
-        return (TADDR) m_pMDImport_UseAccessor;
+
+        // TODO: we used to return "m_pMDImport_UseAccessor" here, - a field that was never initialized to anything.
+        //       we should verify whether this codepath is actually reachable.
+        // return (TADDR)m_pMDImport_UseAccessor;
```

Comment:
This code is weird, but should be left as it was. 

We rename PEFile::m_pMDImport to m_pMDImport_UseAccessor when compiling for the DAC, because we want it to be a compile error if anybody tries to use m_pMDImport in the DAC, since m_pMDImport is the TADDR (debuggee side address) and not a PTR_* that knows how to magically read the value from the debuggee.

However, this one method we actually want to read the TADDR value , do we use m_pMDImport_UseAccessor directly in the DAC code. The code in DacDbiInterfaceImpl::GetPEFileMDInternalRW calls GetMDInternalRWAddress, which will read the right value since m_pMDImport_UseAccessor has the same type and offset as m_pMDImport

Reply:
Thanks @davmason!! I think I got it. I also think this deserves a better comment. It is fairly hard to figure where and how this field is used by just searching around.
I will try adding your comment there as well - at least it would hint that there is more going on than a textual search might reveal.